spring-cloud-modules/spring-cloud-open-telemetry/spring-cloud-open-telemetry1/src/main/java/com/baeldung/opentelemetry/api/client/PriceClient.java

```diff
@@ -0,0 +1,30 @@
+package com.baeldung.opentelemetry.api.client;
+
+import com.baeldung.opentelemetry.model.Price;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class PriceClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(PriceClient.class);
+
+    @Autowired
+    private RestTemplate restTemplate;
```

Comment:
could be `final` and injected by constructor

Reply:
done as mentioned in all places